packages/Telephony/Actions/TerminationCharacterBatchInput.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property.
+    /// </summary>
+    public class TerminationCharacterBatchInput : RegexAggregatorInput
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public new const string Kind = "Microsoft.Telephony.TerminationCharacterBatchInput";
+
+        private string _terminationCharacter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TerminationCharacterBatchInput"/> class.
+        /// </summary>
+        /// <param name="regexAggregatorInput">Implementation for aggregatingInput.</param>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public TerminationCharacterBatchInput([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the character that will be used to signal that the batch of input is complete.
+        /// </summary>
+        [JsonProperty("terminationCharacter")]
+        public string TerminationCharacter
```

Comment:
Do you intend this to be a single character or a multi-char string?

Reply:
Hahahaha I'm still not decided (as you can see from all of the places where I used both string and character) :D Just to avoid regex injection nonsense I'm probably going to convert it to only support one character.

Reply:
Made the user facing description consistent